src/EasyAbp.BookingService.Application.Contracts/EasyAbp/BookingService/AssetOccupancies/Dtos/CreateAssetOccupancyDto.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using Volo.Abp.ObjectExtending;
+
+namespace EasyAbp.BookingService.AssetOccupancies.Dtos;
+
+[Serializable]
+public class CreateAssetOccupancyDto : ExtensibleObject
+{
+    // TODO create by categroy id
+    public Guid? AssetId { get; set; }
+    
+    public Guid? CategoryId { get; set; }
```

Comment:
I think one api is enough, and I will add validation when implementing create method.

Reply:
> I will add validation when implementing create method.

Check it with DTO's validation method so that you don't need to check it in the AppService again. That makes the latter simple and clean.

Reply:
I prefer to create an extra `CreateByCategoryIdAsync` method since people don't need to think about why the `CreateAsync` method has both `AssetId` and `CategoryId`.

> AssetId and CategoryId should not both be empty or both be not empty. Is this validation needed?

Then we don't need to do the empty check anymore.